src/api/searchApi.ts

```diff
@@ -94,11 +94,29 @@ export const fetchNviCandidate = async (identifier: string) => {
   return getNviCandidates.data;
 };
 
-export const fetchResults2 = async (results: number, from: number, { doi = '' }) => {
+interface FetchResultsQuery {
+  doi?: string;
+  identifier?: string;
+  contributor?: string;
+  query?: string;
+  category?: PublicationInstanceType[];
+}
+
+export const fetchResults2 = async (
+  results: number,
+  from: number,
+  { doi, identifier, contributor, query, category }: FetchResultsQuery
+) => {
   const paginationQuery = `results=${results}&from=${from}`;
   const doiQuery = doi ? `doi="${doi}"` : '';
-
-  const fullQuery = [paginationQuery, doiQuery].filter(Boolean).join('&');
+  const idQuery = identifier ? `id="${identifier}"` : '';
+  const contributorQuery = contributor ? `contributor="${contributor}"` : '';
+  const freetextQuery = query ? `query="${query}"` : '';
+  const categoryQuery = category ? `category="${category.join(',')}"` : '';
```

Comment:
virker ikke som ventet. må avklares

Reply:
Update: virker for en kategori nå jeg tok bort `"`. Mangler fortsatt støtte for å søke etter flere